cpp/CMakeLists.txt

```diff
@@ -55,32 +55,36 @@ rapids_find_package(
   INSTALL_EXPORT_SET kvikio-exports
 )
 
-rapids_find_package(
-  cuFile
-  BUILD_EXPORT_SET kvikio-exports
-  INSTALL_EXPORT_SET kvikio-exports
-)
-if(NOT cuFile_FOUND)
-  message(WARNING "Building KvikIO without cuFile")
-else()
-  file(READ "${cuFile_INCLUDE_DIRS}/cufile.h" CUFILE_H_STR)
-  string(FIND "${CUFILE_H_STR}" "cuFileBatchIOSetUp" cuFileBatchIOSetUp_location)
-  if(cuFileBatchIOSetUp_location EQUAL "-1")
-    set(cuFile_BATCH_API_FOUND FALSE)
-  else()
-    set(cuFile_BATCH_API_FOUND TRUE)
-  endif()
-  message(STATUS "Found cuFile Batch API: ${cuFile_BATCH_API_FOUND}")
-  string(FIND "${CUFILE_H_STR}" "cuFileReadAsync" cuFileReadAsync_location)
-  if(cuFileReadAsync_location EQUAL "-1")
-    set(cuFile_STREAM_API_FOUND FALSE)
+if(CUDAToolkit_FOUND)
+  rapids_find_package(
+    cuFile
+    BUILD_EXPORT_SET kvikio-exports
+    INSTALL_EXPORT_SET kvikio-exports
+  )
+  if(NOT cuFile_FOUND)
+    message(WARNING "Building KvikIO without cuFile")
```

Comment:
IMO, this is still a bit too verbose. As a user, I don't quite know what to do with this information. So I don't have cuFile, do I want it? If so, how do I get it? And what happens if I don't have it?

Also consider the case where I know I definitely don't want cuFile (for whatever reason), there is no way to silence this warning. Now suppose I want to run my cmake files using `-Werror` (to catch cmake warnings), I have no way of doing that with this warning.

Note that `CUDAToolKit` will automatically [come shipped with cuFile](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#imported-targets).

My suggestion would be either:
- Consider making this a hard error instead (i.e. `REQUIRED`).
- Or giving the user some way to silence the warning (`KvikIO_SILENCE_MISSING_CUFILE` or something like that).

Reply:
I see you point but let's add explicit options in another PR: https://github.com/rapidsai/kvikio/issues/459.
For now, I have added:
```cmake 
message(
  WARNING "Cannot find cuFile - KvikIO will still work but won't use GPUDirect Storage (GDS)"
)
```